File: PY421/code/eseries.f90
PROGRAM eseries

  IMPLICIT NONE
  INTEGER :: n, k
  REAL(8) :: x, ee, e=1, y=1

  WRITE(*,'("Enter x and the maximum degree in the expansion: ")',&
       ADVANCE='NO')
  READ *,x,n

  ee=EXP(x);
  PRINT '("Exact value of exp(x): ",F14.9)',ee

  y=1
  e=1
  DO k=1,n
     y=y*x
     y=y/k
     e=e+y    
  END DO

  PRINT '("Power series value &
       &of exp(x): ",F14.9)',e

END PROGRAM eseries

File: PY421/code/eseries.c
#include<stdio.h>
#include<math.h>

main(){

  int n;
  double x;
  double ee;
  double e=1, y=1;
  int k;

  printf("Enter x and the maximum degree in the expansion: ");
  scanf("%lf%d",&x,&n);

  ee=exp(x);
  printf("Exact value of exp(x): %14.9f\n",ee);

  y=1;
  e=1;
  for(k=1;k<=n;k++){
    y*=x;
    y/=k;
    e+=y;
  }

  printf("Power series value of exp(x): %14.9f\n",e);

}
